### PR TITLE
Buy item GUI (PART 1)

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    adds a GUI for buy utility items and loot boxes. Allows for use to add more items for purchase.

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Comment by killerswin2** [comment]
Screenshot 2022-10-02 214808

**Comment by Lazejun** [comment]
Pretty neat, probably super annyoing todo since Arma UI Fun.

More informations to the individual Items, the MaxLoad of the LTC Crate, amount of Fuel in the Fuel Containers etc.
Maybe even Node Size so people would know what Vehicles could transport it.
This could be helpful if we add more Options so Players would know what they are getting for their Money.

File: A3A/addons/GUI/dialogues/buyItemDialog.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+
```
[the remark this answers is not in the document]
Comment (HakonRydland) [reply A3A/addons/GUI/dialogues/buyItemDialog.hpp:1]:
```cpp
class A3A_BuyItemDialog : A3A_BuyVehicleDialog {
    idd = A3A_IDD_BUYITEMDIALOG;
    onLoad = "[""onLoad""] spawn A3A_fnc_buyItemDialog";
}
```
note that if these two menus remain unerged then you wont even need uniqe idc's for the controls as the other menu would not ever exist at the same time, we also dont use only idc for control retival but do it through the display, so no fear of conflicts

**Review by HakonRydland — CHANGES_REQUESTED** [review]
alot of good stuff, some minor changes needed

**Comment by killerswin2** [comment]
Screenshot 2022-10-03 181402
Screenshot 2022-10-03 181316
Screenshot 2022-10-03 181412
Screenshot 2022-10-03 181421

File: A3A/addons/GUI/Stringtable.xml [hunk 2]
```diff
@@ -220,6 +220,39 @@
...
                 <French>%1 Sièges FFV</French>
             </Key>
+            <Key ID="STR_antistasi_dialogs_buy_item_tooltip">
+                <Original>Buy %1 for %2 %3</Original>
+            </Key>
+            <Key ID="STR_antistasi_dialogs_buy_item_custom_hint_header">
+                <Original>Purchase Item</Original>
+            </Key>
+            <Key ID="STR_antistasi_dialogs_buy_item_custom_hint_commander_only">
+                <Original>The Commander may only buy this item.</Original>
```
Comment (Bob-Murphy) [inline A3A/addons/GUI/Stringtable.xml:230]: --> "Only the Commander may buy this item"

File: A3A/addons/GUI/Stringtable.xml [hunk 3]
```diff
@@ -220,6 +220,39 @@
...
+                <Original>Loot Boxes</Original>
+            </Key>
+            <Key ID="STR_antistasi_dialogs_vehicle_tab_other">
+                <Original>Other</Original>
+            </Key>
+            <Key ID="STR_antistasi_dialogs_buy_vehicle_refuel_tooltip">
+                <Original>This item can refuel others</Original>
+            </Key>
+            <Key ID="STR_antistasi_dialogs_buy_vehicle_loot_tooltip">
+                <Original>This item can loot items</Original>
```
Comment (Bob-Murphy) [inline A3A/addons/GUI/Stringtable.xml:248]: --> "This item can collect loot" sounds somewhat better than using 2x items

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]

File: A3A/addons/core/functions/init/fn_initClient.sqf [hunk 4]
```diff
@@ -388,23 +388,30 @@ vehicleBox addAction ["Buy Vehicle", {
...
+	};
+	if (isClass (configFile/"CfgVehicles"/_fuelTank # 0)) then {
+		private _dispName = getText (configFile/"CfgVehicles"/_fuelTank # 0/"displayName");
+		vehicleBox addAction [format["Buy %1 for %2€",_dispName, _fuelTank # 1], {[player, _this # 3 # 0, _this # 3 # 1, [['A3A_fnc_initMovableObject', true], ['A3A_fnc_logistics_addLoadAction', false]]] call A3A_fnc_buyItem},_fuelTank,0,false,true,"","_this == theBoss",4];
+	};
+	call A3A_fnc_dropObject;
+	
+	if (LootToCrateEnabled) then {
+		vehicleBox addAction ["Buy loot box for 10€", {player call A3A_fnc_spawnCrate},nil,0,false,true,"","true", 4];
+		call A3A_fnc_initLootToCrate;
```
Comment (HakonRydland) [inline A3A/addons/core/functions/init/fn_initClient.sqf:413]: initLootToCrate called twice

**Review by HakonRydland — APPROVED** [review]
one small mistake, but as initLootToCrate has doubling safeguards its fairly inexpensive and everything else looks ok

File: A3A/addons/core/functions/init/fn_initClient.sqf (hunk 4 above)
Comment (jaj22) [reply A3A/addons/core/functions/init/fn_initClient.sqf:413]: Also dropObject called twice (doesn't matter, but unnecessary) and tab spacing in what's now a 4-spaced file.

Comment (killerswin2) [reply A3A/addons/core/functions/init/fn_initClient.sqf:413]: oof, bad merge

**Comment by killerswin2** [comment]
okay removed double calls, and fixed tabs. Tabs should be spaces.

**Comment by Bob-Murphy** [comment]
Price of the light 10€ - substracted money 25€
all other prices are correct

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Pull request merged**